### PR TITLE
Escape character default bug

### PR DESCRIPTION
**Pull request by PieterOlivier**
When a `\` was encountered followed by a character that has no special meaning, the character was added as an integer.
For instance: `\$` was turned into the string `'36'` instead of the string `'$'`.

### CODE REVIEW

File: src/test/java/io/usethesource/vallang/io/StandardTextReaderTest.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import java.io.StringReader;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+import io.usethesource.vallang.IString;
+import io.usethesource.vallang.IValueFactory;
+import io.usethesource.vallang.ValueProvider;
+import io.usethesource.vallang.type.TypeFactory;
+import io.usethesource.vallang.type.TypeStore;
+
+class StandardTextReaderTest {
```
Comment (DavyLandman) [inline src/test/java/io/usethesource/vallang/io/StandardTextReaderTest.java:15]: Maybe it can move into `RegressionTests`?

**Comment by jurgenvinju** [comment]
Nice catch @PieterOlivier 👍 ; Saved is all from a long debugging session

**Comment by DavyLandman** [comment]
> Saved us all from a long debugging session

well, Pieter was already stuck in stuck in such a session ;)

**Comment by DavyLandman** [comment]
Thanks @PieterOlivier for also fixing the test that was unrelated to your specific fix.

**Pull request merged**